crates/adapters/src/transport/kafka/ft/input.rs

```diff
@@ -312,7 +331,7 @@ impl KafkaFtInputReaderInner {
         while let Some((metadata, ())) = command_receiver.blocking_recv_replay()? {
             let metadata = metadata.parse(n_partitions)?;
             let mut incomplete_partitions = HashSet::new();
-            for (partition, (offsets, receiver)) in
+            for (partition, (offsets, (_, receiver))) in
                 metadata.iter().zip(receivers.iter_mut()).enumerate()
```

Comment:
I think this is wrong, because `incomplete_partitions` is indexed by the partition number, not the index into `partitions`. So, I think that it should be more like this:

```
for (offsets, (partition, receiver)) in metadata.iter().zip(receivers.iter_mut())
```